frontend/public/components/monitoring/dashboards/graph.tsx

```diff
@@ -1,36 +1,69 @@
 import * as React from 'react';
 import { connect } from 'react-redux';
 
+import * as UIActions from '../../../actions/ui';
 import { RootState } from '../../../redux';
 import { FormatLegendLabel, QueryBrowser } from '../query-browser';
 
-const Graph_: React.FC<Props> = ({
+type StateProps = {
+  endTime: number;
+  timespan: number;
+};
+
+type DispatchProps = {
+  setEndTime: (endTime: number) => any;
+  setTimespan: (timespan: number) => any;
+};
+
+type OwnProps = {
+  formatLegendLabel?: FormatLegendLabel;
+  isStack: boolean;
+  pollInterval: number;
+  queries: string[];
+};
+
+type Props = StateProps & DispatchProps & OwnProps;
+
+const Graph: React.FC<Props> = ({
+  endTime,
   formatLegendLabel,
   isStack,
   pollInterval,
   queries,
+  setEndTime,
+  setTimespan,
   timespan,
-}) => (
-  <QueryBrowser
-    defaultSamples={30}
-    formatLegendLabel={formatLegendLabel}
-    hideControls
-    isStack={isStack}
-    pollInterval={pollInterval}
-    queries={queries}
-    timespan={timespan}
-  />
-);
-const Graph = connect(({ UI }: RootState) => ({
-  timespan: UI.getIn(['monitoringDashboards', 'timespan']),
-}))(Graph_);
-
-type Props = {
-  formatLegendLabel?: FormatLegendLabel;
-  isStack: boolean;
-  pollInterval: number;
-  queries: string[];
-  timespan: number;
+}) => {
+  const onZoom = React.useCallback(
+    (from, to) => {
+      setEndTime(to);
+      setTimespan(to - from);
```

Comment:
Since these values are stored in Redux, I'm assuming they will be remembered if I leave the dashboards and come back? I'm trying to decide if that's what we want or if we should try to clear them.

Reply:
Good catch! I think it's better to revert to the default time range when you change dashboard, so I updated it to do that.